src/otx/backend/anomalib/engine.py

```diff
@@ -0,0 +1,117 @@
+"""Anomalib engine."""
+
+# Copyright (C) 2025 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from anomalib.data import AnomalibDataModule
+from anomalib.deploy import ExportType
+from anomalib.engine import Engine as _AnomalibEngine
+from anomalib.models.components import AnomalyModule
+
+from otx.data import OTXPredItem
+from otx.engine import Engine
+from otx.types import ANNOTATIONS, METRICS
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+
+class AnomalyEngine(Engine):
+    """Anomalib engine.
+
+    Args:
+        model: The model to train.
+        data: The data to train on.
+        work_dir: The directory to save the results.
+        **kwargs: Additional keyword arguments for the Anomalib engine. These can also be arguments to the
+             Lightning Trainer.
+
+    Example:
+        >>> from anomalib.models import Padim
+        >>> from anomalib.data import MVTecAD
+        >>> from otx.backend.anomalib.engine import AnomalyEngine
+        >>> engine = AnomalyEngine(model=Padim(), data=MVTecAD())
+        >>> engine.train()
+    """
+
+    def __init__(
+        self,
+        model: AnomalyModule,
+        data: AnomalibDataModule | Path | str,
+        work_dir: Path | str = "results",
+        **kwargs,
+    ):
+        self.model = model
+        self.data = data
+        self.work_dir = work_dir
+
+    def train(self, **kwargs) -> METRICS:
+        """Train the model.
+
+        This runs fit followed by test.
+
+        Returns:
+            METRICS: Dictionary of metrics.
+        """
+        engine = self._create_anomalib_engine(**kwargs)
+        engine.train(model=self.model, datamodule=self.data)
+        # TODO(ashwinvaidya17): return metrics
+        return [{}]
```

Comment:
Anomalib's train endpoint does not return metrics. We need to update anomalib first before exposing this